libs/wingc/src/type_check.rs

```diff
@@ -1814,6 +1845,9 @@ pub struct TypeChecker<'a> {
 	ctx: VisitContext,
 }
 
+/// Class counter, used to generate unique names class types
+static CLASS_COUNTER: AtomicUsize = AtomicUsize::new(0);
```

Comment:
(idea) maybe start the counter at 1 since it looks like all imported JSII classes will have a UID of 0

Reply:
:+1: for consistency I changed the JSII classes to also have a unique id

Reply:
> 👍 for consistency I changed the JSII classes to also have a unique id

Reverted that since it caused snapshot noise. I think we can avoid the class UID altogether by using the (currently optional) FQN field of classes. This seems ideal to me, but I'll need to talk to you to find how to implement this given you're the "bring" expert.